library/include/rocwmma/internal/types_ext.hpp

```diff
@@ -68,43 +100,84 @@ namespace rocwmma
 
     __host__ inline hfloat16_t operator+(const hfloat16_t& x, const hfloat16_t& y)
     {
+#if !defined(__HIP_NO_HALF_OPERATORS__)
         return static_cast<hfloat16_t>(static_cast<float16_t>(x) + static_cast<float16_t>(y));
+#else
+        float16_t fp16_xy = (*(float16_t *)(&x)) + (*(float16_t *)(&y));
+        detail::Fp16Bits fp16(fp16_xy);
+        return fp16.h16;
+#endif
```

Comment:
In both of these cases, operands are converted / interpreted as float16_t. Use the new handy convert function instead and this would simplify to:
```
return convert<hfloat16_t>(convert<float16_t>(x) + convert<float16_t>(y));
```
The convert function takes care of the added case of __ HIP_NO_HALF_CONVERSIONS __

Reply:
This can be applied to simplify all the operators functions below.